Icon preview dialog

jacortinas opened the pull request:
## Description
This PR adds new help menu item and dialog that shows all of the available Octicons and sizes within the app. This is useful for testing icon rendering and referencing during development of features. The menu item is only available in dev and test build of the app.

This changes the global styling for Octicons to stop constraining them to 16px tall. With that global style all of the icons on the preview were limited to 16px tall. After removing the style, all of the icons in the app were still rendered as expected due to 16px being the default height and the height attribute now being specified on the output svg tags.

### Screenshots

![Screenshot of help menu items with show popup octicons selected](https://github.com/desktop/desktop/assets/171215/460e95ff-8187-46e6-ad9d-6406177fb639)
![Screenshot of icon preview dialog](https://github.com/desktop/desktop/assets/171215/68356868-c9e4-4fef-93d1-1d2960867172)


## Release notes

<!--
You can leave this blank if you're not sure.
If you don't believe this PR needs to be mentioned in the release notes, write "Notes: no-notes".
-->

Notes:

jacortinas committed:
Add icon preview dialog

jacortinas committed:
Stop locking icons to 16px tall

koyemyatoo2020 commented:
run

koyemyatoo2020 commented:
> ## ဖော်ပြချက်
> ဤ PR သည် အက်ပ်အတွင်း ရရှိနိုင်သော Octicons နှင့် အရွယ်အစားအားလုံးကို ပြသသည့် အကူအညီမီနူးအသစ်နှင့် ဒိုင်ယာလော့ခ်ကို ပေါင်းထည့်သည်။ ၎င်းသည် အင်္ဂါရပ်များ ဖွံ့ဖြိုးတိုးတက်နေချိန်တွင် အိုင်ကွန် တင်ဆက်ခြင်းနှင့် ကိုးကားခြင်းတို့ကို စမ်းသပ်ရန်အတွက် အသုံးဝင်သည်။ မီနူးကို အက်ပ်၏ dev နှင့် စမ်းသပ်တည်ဆောက်မှုတွင်သာ ရနိုင်သည်။
> 
> ၎င်းသည် ၎င်းတို့အား 16px အရပ်ရှည်စေရန် ကန့်သတ်ထားခြင်းကိုရပ်တန့်ရန် Octicons အတွက် ကမ္ဘာလုံးဆိုင်ရာပုံစံကို ပြောင်းလဲစေသည်။ ထိုကမ္ဘာလုံးဆိုင်ရာပုံစံဖြင့် အစမ်းကြည့်ခြင်းရှိ အိုင်ကွန်အားလုံးကို အမြင့် 16px ကန့်သတ်ထားသည်။ စတိုင်ကို ဖယ်ရှားပြီးနောက်၊ 16px သည် ပုံသေအမြင့်နှင့် အမြင့် attribute ကို output svg တဂ်များတွင် သတ်မှတ်ထားသောကြောင့်၊ အက်ပ်အတွင်းရှိ အိုင်ကွန်အားလုံးကို မျှော်လင့်ထားသည့်အတိုင်း ပြန်ဆိုနေဆဲဖြစ်သည်။
> 
> ### ဖန်သားပြင်ဓာတ်ပုံများ
> ![2024-02-22 ညနေ 3 52 53 နာရီ မျက်နှာပြင်ဓာတ်ပုံ](https://private-user-images.githubusercontent.com/171215/307163898-460e95ff-8187-46e6-ad9d-6406177fb639.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg3MDQyNjQsIm5iZiI6MTcwODcwMzk2NCwicGF0aCI6Ii8xNzEyMTUvMzA3MTYzODk4LTQ2MGU5NWZmLTgxODctNDZlNi1hZDlkLTY0MDYxNzdmYjYzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyM1QxNTU5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZWUxOWY1YzI1YmMzM2Q3YTNkMjBhMGRiMGQzM2JiMDZkYzA5NzdiZTkxM2QxNGU1NzYxNWM4OGRmN2E1ZGYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.e3Nv_cPWP2XINzjO8jVr4JGNKM541Zbap4EvFmz6KSI) ![2024-02-22 ညနေ 3 52 12 နာရီ မျက်နှာပြင်ဓာတ်ပုံ](https://private-user-images.githubusercontent.com/171215/307163932-cc381333-7c40-4b1e-afd9-a0b49752c339.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg3MDQyNjQsIm5iZiI6MTcwODcwMzk2NCwicGF0aCI6Ii8xNzEyMTUvMzA3MTYzOTMyLWNjMzgxMzMzLTdjNDAtNGIxZS1hZmQ5LWEwYjQ5NzUyYzMzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyM1QxNTU5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OGRkODI0MjgyMzg2YzY1ZDVkOTM2NWM0ZTM2MzlmNGFhNzEwYWM2ZWNlYmJmZDdkYWNkZWUwNWQ2MmMxYzVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eCneEsw2LfKMhljKhTdOKw3Wk41vjP-PusGAQQNH7mQ)
> 
> ## ထုတ်ဝေမှုမှတ်စုများ
> မှတ်စုများ-

run

koyemyatoo2020 commented:
ပေါင်းစည်း

koyemyatoo2020 commented:
pull request

jacortinas committed:
Merge branch 'development' into jc-icon-preview-dialog

tidy-dev commented on app/src/ui/octicons/icon-preview-dialog.tsx:
```diff
@@ -0,0 +1,56 @@
+  IIconPreviewDialogProps,
+  {}
+> {
+  public render() {
```
```suggestion
export class IconPreviewDialog extends React.Component<IIconPreviewDialogProps> {
  public render() {
```

Nit: The second parameter is not required

jacortinas committed:
Update app/src/ui/octicons/icon-preview-dialog.tsx

Co-authored-by: tidy-dev <75402236+tidy-dev@users.noreply.github.com>

tidy-dev reviewed (changes requested):
Works great! Only thing I was thinking is the only way to know the sizes is by hovering with mouse, wondering what you think about putting the size under each icon so all the info is accessible?

jacortinas committed:
Update icon preview markup and css

jacortinas commented:
@tidy-dev :wave: I've added the size under each icon preview. I've also simplified the markup to unordered lists inside of an unordered list and used that for CSS targeting.

tidy-dev reviewed (approved):
💖  Thank you for those changes!
